File: C#/2019-2-C샵-10주차-배열길이따기-김유빈/2019-2-C샵-10주차-배열길이따기-김유빈/Form2.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace _2019_2_C샵_10주차_배열길이따기_김유빈
{
    public partial class Form2 : Form
    {
        public Form2()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            if (textBox1.Text == "")
                MessageBox.Show("문장을 입력하시고 버튼을 누르세요", "경고1");
            else
            {
                int iMunjang_Giri;
                iMunjang_Giri = textBox1.TextLength;

                if (iMunjang_Giri % 2 != 0)
                {   //문장길이가 홀수인 경우 Padding작업(무의미한 글자를 추가하는 작업) => 홀수일때는 실행이 되지 않기 때문에 글자추가
                    textBox1.Text = textBox1.Text + "가";
                    iMunjang_Giri = iMunjang_Giri + 1;
                }

                string[] sMunjang_Array = new string[iMunjang_Giri];

                for (int i = 0; i <= iMunjang_Giri - 1; i++)
                {   //문장에서 한글자씩 잘라서 배열로 읽어들어옴.
                    sMunjang_Array[i] = textBox1.Text.Substring(i, 1); 
                }
                textBox2.Text = "";
                for (int i = 0; i <= iMunjang_Giri - 1; i = i + 2)
                {
                    textBox2.Text = textBox2.Text + sMunjang_Array[i + 1] + sMunjang_Array[i]; 
                }
            }
        }

        private void button2_Click(object sender, EventArgs e)
        {
            if (textBox2.Text == "")
                MessageBox.Show("문장을 입력하시고 버튼을 누르세요", "경고2");
            else
            {
                int iMunjang_Giri;
                iMunjang_Giri = textBox2.TextLength;

                //Padding작업 필요 없음. 

                string[] sMunjang_Array = new string[iMunjang_Giri];

                for (int i = 0; i <= iMunjang_Giri - 1; i++)
                {   //문장에서 한글자씩 잘라서 배열로 읽어들어옴.
                    sMunjang_Array[i] = textBox2.Text.Substring(i, 1);
                }
                textBox3.Text = "";
                for (int i = 0; i <= iMunjang_Giri - 1; i = i + 2)
                {
                    textBox3.Text = textBox3.Text + sMunjang_Array[i + 1] + sMunjang_Array[i];
                }
            }
        }
    }
}

File: 비주얼 베이직/기말고사/비주얼베이직-배열/비주얼베이직-배열/Form2.vb
Public Class Form2
    Dim i, j, iMunjang_giri As Integer
    Dim Alphabet_so() = {"a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z"}
    Dim Alphabet_Dae() = {"A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z"}
    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        If TextBox1.Text = "" Then
            MessageBox.Show("영어문장의 소문자로 입력하고 버튼을 누르시오", "경고")
            Exit Sub
        End If
        iMunjang_giri = TextBox1.TextLength
        Dim sMunjang(iMunjang_giri) As String

        For j = 0 To iMunjang_giri - 1
            sMunjang(j) = Mid(TextBox1.Text, j + 1, 1)
        Next
        TextBox2.Text = ""
        For j = 0 To iMunjang_giri - 1
            For i = 0 To 25
                If sMunjang(j) = Alphabet_so(i) Then
                    Exit Sub
                End If
            Next
            If (i + 3) > 25 Then
            Else
                TextBox2.Text = TextBox2.Text & Alphabet_Dae((i + 3) - 26)
            End If
            TextBox2.Text = TextBox2.Text & Alphabet_Dae(i + 3)
        Next
    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        iMunjang_giri = TextBox2.TextLength
        Dim sMunjang(iMunjang_giri) As String
        TextBox3.Text = ""
        For i = 0 To iMunjang_giri - 1
            sMunjang(i) = Mid(TextBox2.Text, i + 1, 1)

        Next
        For j = 0 To iMunjang_giri - 1
            For i = 0 To 25 Step 1
                If sMunjang(j) = Alphabet_Dae(i) Then
                    Exit For
                End If
            Next
            If (i - 3) < 0 Then
                TextBox3.Text = TextBox3.Text & Alphabet_so((i - 3) + 26)
            Else
                TextBox3.Text = TextBox3.Text & Alphabet_so(i - 3)
            End If
        Next
    End Sub
End Class